### PR TITLE
Added add_string_numbers.py solution

### LINKED ISSUE #4: Challenge: add_string_numbers
Add string of numbers and print result

### PR DESCRIPTION
**Pull request by msmsolution**
- [x] PR's title includes the challenge's name and language
- [x] PR has short and clear description of the challenge
- [ ] PR has appropriate labels and milestones for easy identification
- [ ] PR it is assigned to the owner
- [ ] reviewers are assigned
- [ ] the PR contributes only one focused change
- [x] the branch is up to date with `main`/`master`
- [x] the code works when pulled and run locally
- [ ] all conflicts are resolved (if any)
- [x] It is linked to an issue in the appropriate column of the project board
- [ ] feedback is addressed (if any, and if it is appropriate feedback.)

## README Documentation

- [ ] The solution is documented in `/src/README.md`
- [ ] The markdown source is formatted
- [x] Spelling and grammar is correct in all text
- [ ] The markdown looks correct when you preview the file
- [ ] All links and images work
- [ ] The README documents the solution's behavior, strategy(ies) and implementation(s)
- [x] There are use cases in code blocks to illustrate the function's behavior

## Python Files

- [x] There is a module header
- [x] There is a module docstring
- [x] File names are in snake_case
- [x] Test files are named `test_<module_name>.py`

## Function Docstring

- [x] Behavior description
- [x] Parameter description
- [x] Return value description
- [ ] Include any assumptions
- [x] Include 3 or more (passing!) doctests
- [x] Include 1-2 use cases (if necessary)

## Function Implementation

- [x] The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures)
- [x] The code is auto-formatted
- [x] The code has no (reasonable) linting mistakes
- [x] Variables are named with snake_case
- [x] The function has a clear and helpful name
- [x] The file's name matches the function name
- [x] The code follows the strategy as simply as possible
- [x] Variable names are clear and helpful
- [x] Comments explain the strategy (if necessary)
- [x] There are type annotations
- [ ] (challenge) The code includes defensive assertions

## Unit Test Suite

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Each unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases
- [ ] Test suite includes black-box and glass-box tests

### CODE REVIEW

File: src/brain_teasers/python/add_string_numbers/src/add_string_numbers.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#Add string numbers
+#Given two strings of numbers, add the numbers together and return the result as a string.
+#Example:
+#Input: "123", "456"
+
+
+def add_string_numbers(num1, num2):
+    # Fill this in.
+    return str(int(num1) + int(num2))
```
Comment (TasabeehAhmed) [inline src/brain_teasers/python/add_string_numbers/src/add_string_numbers.py:9]: I tested your code and got two errors. Your code only with integers, it doesn't work with floats.
